### PR TITLE
[l10n] Complete Italian (it-IT) locale

### PR DESCRIPTION
**Pull request by mamodev**
I'm an Italian CS student, i hope you will appreciate my small contribution

### CODE REVIEW

**Comment by m4theushw** [comment]
#5467 is merged. Do you want to make improvements to it? Even if no, the translations you sent for the DataGrid are useful.

**Comment by alexfauquette** [comment]
@mamodev I updated your PR to resolve conflicts. If you want to correct some wordings, feel free, otherwise we can merge it as it is right now

**Comment by GhostyJade** [comment]
@mamodev some of the translations string in your latest commit are wrong, can I make some changes to your contribution?

File: packages/grid/x-data-grid/src/locales/itIT.ts [hunk 1]
```diff
@@ -28,16 +28,16 @@ const itITGrid: Partial<GridLocaleText> = {
     count > 1 ? `${count} filtri attivi` : `${count} filtro attivo`,
 
   // Quick filter toolbar field
-  // toolbarQuickFilterPlaceholder: 'Search…',
-  // toolbarQuickFilterLabel: 'Search',
-  // toolbarQuickFilterDeleteIconLabel: 'Clear',
+  toolbarQuickFilterPlaceholder: 'Cerca...',
```
Comment (alexfauquette) [inline packages/grid/x-data-grid/src/locales/itIT.ts:31]:
```suggestion
  toolbarQuickFilterPlaceholder: 'Cerca…',
```

**Comment by alexfauquette** [comment]
@GhostyJade for modification, you can comment line and add suggestions

**Pull request merged**